crates/bevy_asset/src/loader.rs

```diff
@@ -525,6 +568,137 @@ impl<'a> LoadContext<'a> {
+                dependency: path.clone(),
+                error,
+            })?;
+        meta_transform(&mut *meta);
+        self.load_direct_untyped_internal(path.clone(), meta, &*loader, &mut *reader)
+            .await
+    }
+
+    async fn load_direct_with_transform<A: Asset>(
+        &mut self,
+        path: AssetPath<'static>,
+        meta_transform: impl FnOnce(&mut dyn AssetMetaDyn),
+    ) -> Result<LoadedAsset<A>, LoadDirectError> {
+        let (mut meta, loader, mut reader) = self
+            .asset_server
+            .get_meta_loader_and_reader(&path, Some(TypeId::of::<A>()))
+            .await
+            .map_err(|error| LoadDirectError {
+                dependency: path.clone(),
+                error,
+            })?;
+        meta_transform(&mut *meta);
+        self.load_direct_internal(path.clone(), meta, &*loader, &mut *reader)
+            .await
+    }
+
+    /// Loads the asset at the given `path` directly. This is an async function that will wait until the asset is fully loaded before
+    /// returning. Use this if you need the _value_ of another asset in order to load the current asset. For example, if you are
+    /// deriving a new asset from the referenced asset, or you are building a collection of assets. This will add the `path` as a
+    /// "load dependency".
+    ///
+    /// If the current loader is used in a [`Process`] "asset preprocessor", such as a [`LoadTransformAndSave`] preprocessor,
+    /// changing a "load dependency" will result in re-processing of the asset.
+    ///
+    /// [`Process`]: crate::processor::Process
+    /// [`LoadTransformAndSave`]: crate::processor::LoadTransformAndSave
+    pub async fn load_direct<'b, A: Asset>(
+        &mut self,
+        path: impl Into<AssetPath<'b>>,
+    ) -> Result<LoadedAsset<A>, LoadDirectError> {
+        self.load_direct_with_transform(path.into().into_owned(), |_| {})
+            .await
+    }
+
+    /// Loads the asset at the given `path` directly. This is an async function that will wait until the asset is fully loaded before
+    /// returning. Use this if you need the _value_ of another asset in order to load the current asset. For example, if you are
+    /// deriving a new asset from the referenced asset, or you are building a collection of assets. This will add the `path` as a
+    /// "load dependency".
+    ///
+    /// If the current loader is used in a [`Process`] "asset preprocessor", such as a [`LoadTransformAndSave`] preprocessor,
+    /// changing a "load dependency" will result in re-processing of the asset.
+    ///
+    /// If the settings type `S` does not match the settings expected by `A`'s asset loader, an error will be printed to the log
+    /// and the asset load will fail.
+    ///
+    /// [`Process`]: crate::processor::Process
+    /// [`LoadTransformAndSave`]: crate::processor::LoadTransformAndSave
+    pub async fn load_direct_with_settings<'b, A: Asset, S: Settings + Default>(
```

Comment:
Future Work: The `LoadContext` _really_ needs a builder API:

- `load`
- `load_untyped`
- `load_with_settings`
- `load_direct`
- `load_direct_untyped`
- `load_direct_with_reader`
- `load_direct_with_settings`
- `load_direct_untyped_with_reader`
- `load_direct_with_reader_and_settings`

I feel like calling `context.loader().direct().typed::<A>().with_settings(...).load(...)` is much friendlier to code completion, and would allow for documentation on each of the individual options in the API.